File: listas/src/main/java/com/victorlh/android/framework/listas/adapter/ListaObservable.java
package com.victorlh.android.framework.listas.adapter;

import androidx.annotation.NonNull;

import com.victorlh.android.framework.listas.ItemLista;

public abstract class ListaObservable<T extends ItemLista> {

	protected void setItem(T item) {
	}

	protected void deleteItem(T item) {
	}

	protected void recargarLista() {
	}

	protected boolean onClickElemento(AbstractViewHolder<T> viewHolder) {
		return false;
	}

	protected boolean onLongClickElemento(AbstractViewHolder<T> viewHolder) {
		return false;
	}

	protected void onBindViewHolder(@NonNull final AbstractViewHolder<T> holder, int position) {
	}
}

File: listas/src/main/java/com/victorlh/android/framework/listas/adapter/ListaObservable.kt
package com.victorlh.android.framework.listas.adapter

abstract class ListaObservable<T> {

    open fun setItem(item: T) {}
    open fun deleteItem(item: T) {}
    open fun recargarLista() {}
    open fun onClickElemento(viewHolder: AbstractViewHolder<T>): Boolean = false
    open fun onLongClickElemento(viewHolder: AbstractViewHolder<T>): Boolean = false
    open fun onBindViewHolder(holder: AbstractViewHolder<T>, position: Int) {}
}